src/lib.rs

```diff
@@ -0,0 +1,64 @@
+//#![cfg_attr(not(feature = "std"), no_std)]
+use ed25519_dalek::{VerifyingKey, Signature};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+
+#[wasm_bindgen]
+pub fn verify_ed25519(
+    data: &[u8], // [key (32 bytes), signature (64 bytes), message (? bytes); item_length]
+    item_length: usize,
+) -> Vec<u8> {
+    let results = data.chunks(item_length).map(|chunk| {
+        let maybe_key = &chunk[0..32].try_into();
+        let maybe_signature = &chunk[32..96].try_into();
```

Comment:
This can panic when `chunk` does not have enough bytes. Maybe check that `item_length` is greater than 96 (or 97? can message be empty?) and return empty vector in response or sth.

Reply:
fixed